internal/declarative/v2/reconciler.go

```diff
@@ -366,7 +366,12 @@ func (r *Reconciler) renderTargetResources(
 func (r *Reconciler) pruneDiff(
 	ctx context.Context, clnt Client, obj Object, renderer Renderer, diff []*resource.Info,
 ) error {
-	if err := r.deleteResources(ctx, clnt, obj, pruneKymaSystem(diff)); err != nil {
+	diff = pruneResource(diff, "Namespace", namespaceNotBeRemoved)
+	resourceName := r.ModuleCRDName(obj)
+	if resourceName != "" {
```

Comment:
I think you don't need this additional check if resourceName is empty, `pruneResource` anyway will skip this case.